src/gretel_synthetics/config.py

```diff
@@ -64,3 +65,13 @@ def _set_tokenizer(self):
         self.tokenizer_prefix = "m"
         self.tokenizer_model = Path(self.checkpoint_dir, 'm.model').as_posix()
         self.training_data = Path(self.checkpoint_dir, 'training_data.txt').as_posix()
+
+    def as_dict(self):
+        config_dict = {key: value for key, value in self.__dict__.items()
+                       if not key.startswith('_') and not callable(key)}
+
+        # delete temporary configuration items
+        for key in ['tokenizer_prefix', 'tokenizer_model', 'training_data', 'tokenizer', 'processed_data']:
+            del config_dict[key]
+
+        return config_dict
```

Comment:
We should probably look to make the config base class a dataclass. This will give us more natural serialization. I can take a look at this if you like